feat: add delivery status update functionality

abhimanyurajeesh opened the pull request:
## Proposed Changes

fixes: #14399


Added delivery status update functionality using dropdown menu for actions in SupplyDeliveryTable and DeliveryOrderShow

<img width="302" height="188" alt="image" src="https://github.com/user-attachments/assets/964bef5c-d3ec-4ed0-abd1-63dd35d90528" />

<img width="860" height="464" alt="image" src="https://github.com/user-attachments/assets/baa6e188-f52d-4150-a0dc-68dd8a063f20" />


<img width="904" height="191" alt="image" src="https://github.com/user-attachments/assets/935e8300-9814-48c3-89ea-9739915dcbbf" />

Tagging: @ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate the bug or test the new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [x] Ensure that UI text is placed in I18n files.
- [x] Prepare a screenshot or demo video for the changelog entry and attach it to the issue.
- [x] Request peer reviews.
- [x] Complete QA on mobile devices.
- [x] Complete QA on desktop devices.


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Mark delivery orders or individual deliveries as "entered in error" or "abandoned" from dropdown actions with confirmation dialogs and localized success toasts; list refreshes and actions disable while processing. Draft orders show contextual actions.

* **Refactor**
  * Deliveries table redesigned with a modular layout and expanded columns (items, requested/received quantities, dates, monetary breakdowns, status, condition); per-row UI updated.

* **Documentation**
  * Added confirmation descriptions and updated toast message text for status changes.

* **Tests**
  * Updated test expectation wording for the status-change toast.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

abhimanyurajeesh committed:
feat: add delivery status update functionality and dropdown menu for actions in SupplyDeliveryTable and DeliveryOrderShow

abhimanyurajeesh committed:
remove -

abhimanyurajeesh commented on src/pages/Facility/services/inventory/externalSupply/deliveryOrder/DeliveryOrderShow.tsx:
```diff
@@ -1,10 +1,17 @@
+  ChevronLeft,
+  Edit,
+  EllipsisVertical,
+  MoreVertical,
```
both are used

abhimanyurajeesh committed:
feat: add deliveryOrderStatus prop to SupplyDeliveryTable and pass status from DeliveryOrderShow

abhimanyurajeesh committed:
Merge branch 'develop' into manyu/feat/delivery-status-update

abhimanyurajeesh committed:
Merge branch 'develop' into manyu/feat/delivery-status-update

abhimanyurajeesh committed:
Merge branch 'develop' into manyu/feat/delivery-status-update

abhimanyurajeesh committed:
Merge remote-tracking branch 'upstream/develop' into manyu/feat/delivery-status-update

bodhish commented on src/pages/Facility/services/inventory/externalSupply/deliveryOrder/DeliveryOrderShow.tsx:
```diff
@@ -435,6 +443,48 @@ export function DeliveryOrderShow({
                 </Button>
               )}
+
+            {deliveryOrder.status === DeliveryOrderStatus.draft && (
```
Both of these are destructive action that requires additional confirmation. 

CC @nihal467

abhimanyurajeesh replied:
Done, will add. But there no direct click so should we add !?

abhimanyurajeesh committed:
Merge branch 'develop' into manyu/feat/delivery-status-update

abhimanyurajeesh committed:
fix: update identifier search configuration in PatientIndex component (#14264)

abhimanyurajeesh committed:
feat: add confirmation dialogs and update translations for delivery order status changes

- Implemented confirmation dialogs for marking delivery orders as entered in error and abandoned.
- Added corresponding success messages and confirmation descriptions in the translation file.
- Updated UI components to utilize the new confirmation dialog for status updates.

abhimanyurajeesh commented on src/pages/Facility/services/inventory/SupplyDeliveryTable.tsx:
```diff
@@ -43,9 +58,39 @@ export function SupplyDeliveryTable({
+    onSuccess: () => {
+      queryClient.invalidateQueries({ queryKey: ["supplyDeliveries"] });
+      toast.success(t("status_updated_successfully"));
+    },
```
global error handler ?

abhimanyurajeesh committed:
Merge branch 'develop' into manyu/feat/delivery-status-update

abhimanyurajeesh commented on src/pages/Facility/services/inventory/SupplyDeliveryTable.tsx:
```diff
@@ -76,154 +121,171 @@ export function SupplyDeliveryTable({
+                    </DropdownMenuContent>
+                  </DropdownMenu>
+                </TableCell>
+              )}
```
this is correct as once you update, you shouldn't be able to change the status again

abhimanyurajeesh commented on src/pages/Facility/services/inventory/externalSupply/deliveryOrder/DeliveryOrderShow.tsx:
```diff
@@ -211,28 +228,42 @@ export function DeliveryOrderShow({
+        statusMessages[updatedDeliveryOrder.status] ||
+          t("order_updated_successfully"),
       );
     },
```
we have global error handler that takes care of this

abhimanyurajeesh commented on src/pages/Facility/services/inventory/SupplyDeliveryTable.tsx:
```diff
@@ -43,9 +58,39 @@ export function SupplyDeliveryTable({
+    newStatus: SupplyDeliveryStatus,
+  ) => {
+    updateDeliveryStatus({
+      deliveryId,
```
na

abhimanyurajeesh committed:
refactor: update button classes for consistent styling in DeliveryOrderShow component

- Changed button class names to use 'justify-stretch' for better alignment.
- Wrapped button text in a span for improved semantic structure.

nihal467 committed:
Merge branch 'develop' into manyu/feat/delivery-status-update

nihal467 commented:
LGTM

amjithtitus09 commented on public/locale/en.json:
```diff
@@ -3458,8 +3460,10 @@
   "oral_issue_for_non_oral_nutrition_route_error": "Can be specified only if nutrition route is set to Oral",
   "order": "Order",
   "order_created": "Order created successfully",
+  "order_marked_as_abandoned_successfully": "Order marked as abandoned successfully",
```
Can we reuse these by passing in status?

abhimanyurajeesh replied:
I thought about that but pending status is Approved, so figured out adding this better ?

amjithtitus09 commented on src/pages/Facility/services/inventory/externalSupply/deliveryOrder/DeliveryOrderShow.tsx:
```diff
@@ -211,28 +228,42 @@ export function DeliveryOrderShow({
-            ? t("order_marked_as_completed_successfully")
-            : t("order_updated_successfully"),
+        statusMessages[updatedDeliveryOrder.status] ||
+          t("order_updated_successfully"),
```
pass status to translation

amjithtitus09 commented on src/pages/Facility/services/inventory/externalSupply/deliveryOrder/DeliveryOrderShow.tsx:
```diff
@@ -401,14 +432,15 @@ export function DeliveryOrderShow({
-                </Link>
-              </Button>
-            )}
+            {!isRequester &&
```
May not be your change but this would hide Edit for draft external order right?

abhimanyurajeesh replied:
nope, will it ?

amjithtitus09 commented on src/pages/Facility/services/inventory/SupplyDeliveryTable.tsx:
```diff
@@ -43,9 +58,39 @@ export function SupplyDeliveryTable({
+    },
+  });
+
+  const updateStatus = (
```
Is this wrapper useful?

abhimanyurajeesh committed:
Refactor delivery order status messages and update status handling in SupplyDeliveryTable

abhimanyurajeesh commented on src/pages/Facility/services/inventory/externalSupply/deliveryOrder/DeliveryOrderShow.tsx:
```diff
@@ -1,11 +1,19 @@
+  ChevronLeft,
+  Edit,
+  EllipsisVertical,
+  MoreVertical,
```
its used in line 621

abhimanyurajeesh commented on src/pages/Facility/services/inventory/SupplyDeliveryTable.tsx:
```diff
@@ -1,27 +1,41 @@
   TableHeader,
   TableRow,
-} from "@/components/ui/table";
+} from "@/components/Common/Table";
```
it was a intended change

abhimanyurajeesh commented on src/pages/Facility/services/inventory/SupplyDeliveryTable.tsx:
```diff
@@ -43,6 +56,35 @@ export function SupplyDeliveryTable({
+    onSuccess: () => {
+      queryClient.invalidateQueries({ queryKey: ["supplyDeliveries"] });
+      toast.success(t("status_updated_successfully"));
+    },
```
global error handler will take care

yash-learner commented:
@abhimanyurajeesh 

Can you please file an issue and link to this ?

yash-learner reviewed (commented):
Currently we are not showing the entered in error and abandoned orders  (No tabs)

are we planning  to do in a separate PR ?

abhimanyurajeesh committed:
Merge remote-tracking branch 'upstream/develop' into manyu/feat/delivery-status-update

abhimanyurajeesh committed:
changed font size to match figma

abhimanyurajeesh committed:
fix playwright

abhimanyurajeesh committed:
fix the cell

Jacobjeevan commented on public/locale/en.json:
```diff
@@ -2873,6 +2873,8 @@
   "mark_encounter_as_complete_confirmation": "Are you sure you would like to mark this encounter as complete?",
   "mark_for_discharge": "Mark for discharge",
   "mark_inactive": "Mark Inactive",
+  "mark_order_as_abandoned_confirmation_description": "Are you sure you want to mark this delivery order as abandoned? This action indicates that the order was discontinued before completion. This action cannot be undone.",
```
we can combine these two no?

Jacobjeevan commented on src/pages/Facility/services/inventory/externalSupply/deliveryOrder/DeliveryOrderShow.tsx:
```diff
@@ -197,28 +214,30 @@ export function DeliveryOrderShow({
 
-  const { mutate: updateDeliveryOrder, isPending: isUpdating } = useMutation({
+  const { mutate: updateDeliveryOrder, isPending: isUpdating } = useMutation<
+    DeliveryOrderRetrieve,
```
is this needed? 🤔  should get from API no?

Jacobjeevan reviewed (approved):
lgtm
